core/common/util/src/test/java/org/eclipse/dataspaceconnector/common/reflection/ReflectionUtilTest.java

```diff
@@ -195,4 +196,33 @@ void getFieldValue_withArrayIndex_outOfBounds() {
         var o = new TestObjectWithList("test-desc", 0, List.of(new TestObject("to1", 420), new TestObject("to2", 69)));
         assertThatThrownBy(() -> ReflectionUtil.getFieldValue("nestedObjects[3]", o)).isInstanceOf(IndexOutOfBoundsException.class);
     }
-}
\ No newline at end of file
+
+    @Test
+    void getAllFieldsRecursiveWithPath_whenNotDeclared() {
+        assertThat(ReflectionUtil.getAllFieldsRecursiveWithPath(TestObjectWithList.class, "notExist")).isEmpty();
+    }
+
+    @Test
+    void getAllFieldsRecursiveWithPath() {
+        var fields = ReflectionUtil.getAllFieldsRecursiveWithPath(TestObjectSubSubclass.class, "anotherObject");
+        assertThat(fields).hasSize(1).extracting(Field::getName).containsExactly("anotherObject");
+    }
+
+    @Test
+    void getAllFieldsRecursiveWithPath_whenDottedNotationProvided() {
+        var fields = ReflectionUtil.getAllFieldsRecursiveWithPath(TestObjectWithList.class, "nestedObjects.description");
+        assertThat(fields).hasSize(2).extracting(Field::getName).containsExactly("nestedObjects", "description");
+    }
+
+    @Test
+    void getSuperClassFirstParameterType() {
+        var fields = ReflectionUtil.getSuperClassFirstParameterType(TestGenericSubclass.class);
+        assertThat(fields).isEqualTo(Optional.of(String.class));
+    }
+
+    @Test
+    void getSuperClassFirstParameterType_whenNoGenericSuperclass() {
```

Comment:
In this case i think the generic type is lost, Doing a debug session i got type `A` which is not a class, that's why it's filtered our. Probably it's better to have a local function for this specific task instead of a generic function which could not work always?

Reply:
I've added a test case on this `getSingleSuperTypeGenericArgument_whenGenericClass`. 
Seems that this kind of usage is not supported and the actual type info is lost. 
If you have any hint would be great :)